libs/ipc/inc/public/tfc/ipc/packet.hpp

```diff
@@ -118,7 +120,8 @@ struct packet {
     return {};
   }
 
-  static constexpr auto deserialize(std::ranges::view auto&& buffer) -> std::expected<value_t, std::error_code> {
+  template <typename error_t = std::error_code>
+  static constexpr auto deserialize(std::ranges::view auto&& buffer) -> std::expected<value_t, error_t> {
```

Comment:
This is also unnecessary.
This can be a bit mind boggling, in case of mass the value_t is std::expected so the return type is std::expected<std::expected<mass_t, mass_error_e>, std::error_code>